pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -747,7 +748,7 @@ func (tc *testCase) runTestWithController(t *testing.T, hpaController *Horizonta
 	stop := make(chan struct{})
 	defer close(stop)
 	informerFactory.Start(stop)
-	go hpaController.Run(stop)
+	go hpaController.Run(context.TODO())
```

Comment:
I think the failure is related to this, it seems to be panicking on the calls related to here:

```
E1004 19:10:12.992779   52130 runtime.go:76] Observed a panic: Fail in goroutine after TestUpscaleCap has completed
...
created by k8s.io/kubernetes/pkg/controller/podautoscaler.(*HorizontalController).Run
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/podautoscaler/horizontal.go:177 +0x4bc
```

which is a call to `wait.UntilWithContext`, so maybe `context.TODO()` doesn't work for that? Or I just be using the same context for this and the stop channel